src/jira/client/index.ts

```diff
@@ -35,38 +36,40 @@ async function getJiraClient(
 	const client = {
 		baseURL: instance.defaults.baseURL,
 		issues: {
-			// eslint-disable-next-line camelcase
-			get: (issue_id, query = { fields: "summary" }): Promise<AxiosResponse> =>
+			get: (issueId: string, query = { fields: "summary" }): Promise<AxiosResponse<JiraIssue>> =>
 				instance.get("/rest/api/latest/issue/:issue_id", {
 					urlParams: {
 						...query,
-						issue_id
+						issue_id: issueId
 					}
 				}),
-			getAll: async (issueIds, query) =>
-				(
-					await Promise.all<AxiosResponse>(
-						issueIds.map((issueId) => client.issues.get(issueId, query)
-							// Ignore any errors
-							.catch(() => undefined))
-					)
-				)
-					.filter((response) => response?.status === 200 && response?.data)
-					.map((response) => response.data),
-			parse: (text) => {
-				if (!text) return null;
-				return issueKeyParser().parse(text);
+			getAll: async (issueIds: string[], query?: { fields: string }): Promise<JiraIssue[]> => {
+				const responses = await Promise.all<AxiosResponse<JiraIssue> | undefined>(
+					issueIds.map((issueId) => client.issues.get(issueId, query)
+						// Ignore any errors
+						.catch(() => undefined))
+				);
+				return responses.reduce((acc: JiraIssue[], response) => {
+					if (response?.status === 200 && !!response?.data) {
+						acc.push(response.data);
+					}
+					return acc;
```

Comment:
Why did we have to replace filter.map with reduce?

Reply:
filter and map doesn't have the correct typings as they assume it would have the same typings as the array before it, which filter was actually changing.  Plus, a single reduce function is quicker than to do both filter & map while keeping the typings correct.